### PR TITLE
MINOR: add AUTO_OFFSET_RESET_CONFIG to StreamsConfig,

### PR DESCRIPTION
**Pull request by ymatsuda**
and remove TOTAL_RECORDS_TO_PROCESS
@guozhangwang

### CODE REVIEW

**Comment by guozhangwang** [comment]
Could you update the examples to use StreamsConfig.AUTO_OFFSET_RESET_CONFIG as well?

**Comment by ymatsuda** [comment]
examples are updated.

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 1]
```diff
@@ -117,6 +114,10 @@
     /** <code>client.id</code> */
     public static final String CLIENT_ID_CONFIG = CommonClientConfigs.CLIENT_ID_CONFIG;
 
+    /** <code>auto.offset.reset</code> */
+    public static final String AUTO_OFFSET_RESET_CONFIG = ConsumerConfig.AUTO_OFFSET_RESET_CONFIG;
+    private static final String AUTO_OFFSET_RESET_DOC = "What to do when there is no initial offset in Kafka or if the current offset does not exist any more on the server (e.g. because that data has been deleted): <ul><li>earliest: automatically reset the offset to the earliest offset<li>latest: automatically reset the offset to the latest offset</li><li>none: throw exception to the consumer if no previous offset is found for the consumer's group</li><li>anything else: throw exception to the consumer.</li></ul>";
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: Can we assign to `ConsumerConfig.AUTO_OFFSET_RESET_DOC` as well?

**Comment by guozhangwang** [comment]
LGTM. Merged to trunk.

**Pull request closed without merging**